lib/archethic/p2p/node.ex

```diff
@@ -353,7 +283,8 @@ defmodule Archethic.P2P.Node do
       DateTime.to_unix(enrollment_date)::32, available_bin::1, synced_bin::1, authorized_bin::1,
       authorization_date::32, first_public_key::binary, last_public_key::binary,
       reward_address::binary, last_address::binary, origin_public_key::binary,
-      DateTime.to_unix(last_update_date)::32, DateTime.to_unix(availability_update)::32>>
+      DateTime.to_unix(last_update_date)::32, DateTime.to_unix(availability_update)::32,
+      mining_public_key::binary>>
```

Comment:
According to typespec, mining_public_key can be nil and you cannot encode nil as a binary.
You should have a check before to determine if the value is nil.
Same for deserialize function